### PR TITLE
HSEARCH-1920 documentation updates for HSEARCH-668

### PR DESCRIPTION
**Pull request by Sanne**
*preview* as it requires HSEARCH-668 to be merged first.

It contains all of  HSEARCH-668 as it wasn't merged yet, but please look just at the last commit.

### CODE REVIEW

**Comment by Sanne** [comment]
thanks Emmanuel, sorry that text was quite poor.. I was too rushed to push and distracted :)

**Comment by Sanne** [comment]
Feedback integrated, marked updated now as this is the last blocking piece to release the transactional JMS feature.

File: documentation/src/main/asciidoc/configuration.asciidoc [hunk 1]
```diff
@@ -474,6 +474,10 @@ shown in <<table-worker-configuration>>.
             `hibernate.search.default.worker` are passed to the
             Worker during initialization. This allows adding custom, worker
             specific parameters.
+|hibernate.search.worker.enlist_in_transaction|Defaults to `false`. Set it to `true`
+            to have all indexing work sent to the queue within the same transaction
+            as the Hibernate ORM Sesssion. This options should only be enabled when all
```
Comment (DavideD) [inline documentation/src/main/asciidoc/configuration.asciidoc]: sesSsion

File: documentation/src/main/asciidoc/architecture.asciidoc [hunk 2]
```diff
@@ -98,7 +98,7 @@ All index update operations applied on a given node are sent to a JMS queue. A u
 then process the queue and update the master index. The master index is then replicated on a regular
 basis to the slave copies. This is known as the master/slaves pattern. The master is the sole
 responsible for updating the Lucene index. The slaves can accept read as well as write operations.
-However, they only process the read operation on their local index copy and delegate the update
+However while they process the read operation on their local index copy they will delegate the update
```
Comment (DavideD) [inline documentation/src/main/asciidoc/architecture.asciidoc]: ... the read operation**S** on their local index copy**,**

File: documentation/src/main/asciidoc/architecture.asciidoc [hunk 3]
```diff
@@ -107,6 +107,14 @@ This mode targets clustered environments where throughput is critical, and index
 affordable. Reliability is ensured by the JMS provider and by having the slaves working on a local
 copy of the index.
 
+The JMS integration can be transactional. With this backend (and currently only this backend) you
+can have Hibernate Search send the indexing work into the queue within the same transaction
+applying changes to the relational database. This options requires you to use an XA transaction.
```
Comment (DavideD) [inline documentation/src/main/asciidoc/architecture.asciidoc]: This option (without S)

File: documentation/src/main/asciidoc/architecture.asciidoc [hunk 4]
```diff
@@ -107,6 +107,14 @@ This mode targets clustered environments where throughput is critical, and index
...
+
+By default this backend's transactional capabilities are not enabled, but messages will be
```
Comment (DavideD) [inline documentation/src/main/asciidoc/architecture.asciidoc]: not enabled = disabled

**Comment by DavideD** [comment]
Looks good, I'm going to fix the few typos and integrate it if it's good for you

File: documentation/src/main/asciidoc/architecture.asciidoc (hunk 4 above)
Comment (Sanne) [inline documentation/src/main/asciidoc/architecture.asciidoc]: +1 could you also change
s/, but messages will/: messages will/

**Comment by DavideD** [comment]
Squashed the fixes and merged https://github.com/hibernate/hibernate-search/pull/865

**Pull request closed without merging**